tests/ui_tools/test_boxes.py

```diff
@@ -215,6 +215,36 @@ def test_generic_autocomplete_emojis(self, write_box, text,
         typeahead_string = write_box.generic_autocomplete(text, state)
         assert typeahead_string == required_typeahead
 
+    @pytest.mark.parametrize(['key', 'current_typeahead_mode',
+                              'expected_typeahead_mode',
+                              'expect_footer_was_reset'], [
+        # footer does not reset
+        (keys_for_command('AUTOCOMPLETE').pop(), False, False, False),
+        (keys_for_command('AUTOCOMPLETE_REVERSE').pop(), False, False, False),
+        (keys_for_command('AUTOCOMPLETE').pop(), True, True, False),
+        (keys_for_command('AUTOCOMPLETE_REVERSE').pop(), True, True, False),
+        # footer resets
+        (keys_for_command('GO_BACK').pop(), True, False, True),
+        ('space', True, False, True),
+        ('k', True, False, True),
+    ])
+    def test_keypress_typeahead_mode_autocomplete_key(self, mocker, write_box,
+                                                      current_typeahead_mode,
+                                                      expected_typeahead_mode,
+                                                      expect_footer_was_reset,
+                                                      key):
+
+        self.view.set_footer_text = mocker.patch(
+                                    'zulipterminal.ui.View.set_footer_text')
```

Comment:
Is the same simplification possible here, as discussed elsewhere? Can we just remove this line, since the entire view is mocked already?